File: applicatie/src/main/java/com/vdzon/administratie/rekening/RekeningResource.java
package com.vdzon.administratie.rekening;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;


public class RekeningResource {

    @Inject
    public RekeningResource(RekeningService rekeningService) {
        Spark.put("/rest/rekening/", rekeningService::putRekening, JsonUtil.json());
        Spark.post("/rest/rekening/", rekeningService::putRekening, JsonUtil.json());
        Spark.delete("/rest/rekening/:uuid", rekeningService::removeRekening, JsonUtil.json());
    }


}

File: rest/src/main/java/com/vdzon/administratie/rest/rekening/RekeningResource.kt
package com.vdzon.administratie.rest.rekening

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject


class RekeningResource
@Inject
constructor(rekeningService: RekeningService) {
    init {
        Spark.put("/rest/rekening/", Route { req, res -> rekeningService.putRekening(req, res) }, JsonUtil.json())
        Spark.post("/rest/rekening/", Route { req, res -> rekeningService.putRekening(req, res) }, JsonUtil.json())
        Spark.delete("/rest/rekening/:uuid", Route { req, res -> rekeningService.removeRekening(req, res) }, JsonUtil.json())
    }


}
